scene/gui/tree.cpp

```diff
@@ -1784,13 +1784,22 @@ int Tree::propagate_mouse_event(const Point2i &p_pos, int x_ofs, int y_ofs, bool
 					cache.click_type = Cache::CLICK_NONE;
 					return -1;
 				}
+
+				// Make sure the click is correct.
+				Point2 click_pos = get_global_mouse_position() - get_global_position();
+				if (get_item_at_position(click_pos) == NULL) {
```

Comment:
I'm concerned about the potential performance impact, `get_item_at_position` sounds pretty heavy handed. But I don't have a better fix to propose for now.

Reply:
It's one additional call though, also guarded by multiple other conditions. It's called at the same time as the original one (at the time of a click).

But yeah, unexpected things might happen :I

Reply:
Side note, should be updated to `nullptr` now.